File: haskell/eg/left-turner.hs
import Tankode.Basic

ident :: Id
ident = Id
  { name = "left-turner"
  , bodyColour   = "red3"
  , radarColour  = "red3"
  , scanColour   = "red8"
  , trackColour  = "red3"
  , bulletColour = "red9"
  , gunColour    = "yellow7"
  }

leftTurner :: Tankode ()
leftTurner input = output
  { accel = 0
  , body  = 0
  , gun   = maxGunTurn
  , shoot = case enemy input of
              Nothing -> 0
              Just _  -> 2/3
  }

main :: IO ()
main = run ident leftTurner ()

File: c/raw/eg/left-turner.c
#include <tankode.h>

struct tankode_out left_turner(struct tankode_in in)
{
	struct tankode_out out = {0., 0., 0., 0., 0.};
	out.gun   = (2./3.) / 120.;
	out.shoot = in.enemy == in.enemy ? (2./3.) : 0.;
	return out;
}

int main()
{
	struct tankode_id id = {
		"left-turner",
		"orange4",
		"orange5",
		"orange6",
		"orange7",
		"orange8",
		"orange9"
	};
	tankode_run(id, left_turner);
	return 0;
}
